### DISCUSSION TITLE
Legit app version from Play Store not returning LICENSED

### DISCUSSION THREAD

**Discussion by dhamelin** [KirillKuzyk/Google-Play-License-Check#15] [General]
Hello, I have updated my game using your LicenseCheck, it seems to work well for hundreds of players. Until tonight, a "false-positive" seems to have happened. A player shown me proof that the app was downloaded through the Google Play Store, but my code think it's not a legit version. I have this code:

```
void UpdateStatus(LicenseStatus status)
{
    Status = status;
        
    if (status == LicenseStatus.NOT_LICENSED || 
        status == LicenseStatus.ERROR_NOT_MARKET_MANAGED || 
        status == LicenseStatus.ERROR_INVALID_PACKAGE_NAME || 
        status == LicenseStatus.ERROR_NON_MATCHING_UID)
    {
        IsLegit = false;
    } 
}
```

It means that his device returns 1 of the 4 values (probably NOT_LICENSED? I don't know yet.)

Here are his device info:

BlackBerry BBE100-4
Android OS 8.1.0 / API-27 (OPM1.171019.026/ACE748)

I'm wondering if this situation has been seen by some one or where could I find help investigating this.

Thank you!

**Comment by KyryloKuzyk** [comment]
Hi,

Do you have any other places that set the IsLegit flag?

I have one suggestion for investigating this, but it requires uploading a new app version to Google Play. The `GooglePlay.License.debugInfo` string field can help debug licensing issues. You can hide it (and other relative debugging info like licensing status) somewhere inside your app, and if a user encounters a problem, you can ask them to show you the debug info. This solution can be applied not only to my plugin, but to other tools with external dependencies you don't have control over.

As to how to address the issue without uploading a new app version, I suggest you to ask the user to clear the Google Play app's cache and reinstall the app.

**Reply by dhamelin** [reply]
Thanks for the suggestion. 

IsLegit is only set from that method.

I have started to add the licence status to player's profile, which I'm able to see from our server. Not sure if this will help, but I like your debugInfo solution too.

**Comment by dhamelin** [comment]
I have started to receive a bit more details and I'm seeing this:

license.status is: NOT_LICENSED
license.debugInfo is: Status is LICENSED, but message is empty.

Does it mean that we have received a LICENSED response, but it can't be trusted?

**Reply by KyryloKuzyk** [reply]
This is weird; such a situation should never happen under normal circumstances. 
Google Play licensing server should return three fields: status code, message, and the message's signature. If a message or signature is missing, the status code can't be validated, and such a response can't be trusted.

Do you have any additional info that can prove that the user downloaded the app through Google Play and haven't modified the app in any way?
